### ISSUE TITLE
dart protobuf3 json google.brotobuf.any   type_null is null

### ISSUE THREAD

**Issue by zhchzh** [google/protobuf.dart#454]
I want to serialize JSON into protobuf3 object, in which there is any type data, which cannot be serialized successfully. After breakpoint debugging, type_url  is null

```
dependencies: protobuf 1.0.1
protobuf/src/protobuf/mixins/well_known.dart 

static void fromProto3JsonHelper(GeneratedMessage message, Object json,
      TypeRegistry typeRegistry, JsonParsingContext context) {
    if (json is! Map<String, dynamic>) {
      throw context.parseException(
          'Expected Any message encoded as {@type,...},', json);
    }

    final object = json as Map<String, dynamic>;
    final typeUrl = object['@type'];    // This is null  ...............................................  


    if (typeUrl is String) {
      AnyMixin any = message as AnyMixin;
      BuilderInfo info = typeRegistry.lookup(_typeNameFromUrl(typeUrl));
      if (info == null) {
        throw context.parseException(
            'Decoding Any of type ${typeUrl} not in TypeRegistry $typeRegistry',
            json);
      **}**
```

**Comment by sigurdm** [comment]
The json encoding of Any objects is documented here: https://developers.google.com/protocol-buffers/docs/proto3#json

How does your json object look?

Does it have a "@type" entry? Otherwise it would fail the decoding.

 How did you encode it? If you encode it with dart protobuf you need to pass a "TypeRegistry"
See an example of encoding an any type here: https://github.com/dart-lang/protobuf/blob/master/protoc_plugin/test/proto3_json_test.dart#L255

**Comment by osa1** [comment]
Closing due to inactivity. Please feel free to reopen if you still have this problem with the latest release. Latest release includes at least one fix (#568) related to `Any`.
